tgui/packages/tgui/interfaces/PollEditorPanel.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Mordent (https://github.com/mordent-goonstation)
+ * @license ISC
+ */
+
+import { useBackend, useLocalState } from '../backend';
+import { Button, Section, Stack } from '../components';
+import { Window } from '../layouts';
+import { PollEditorForm } from './common/forms/PollEditorForm';
+import type { ExpiryOptions, ExpiryType, PollSettings } from './common/forms/PollEditorForm';
+
+interface PollEditorPanelData {
+  lastError: string | null;
+  serverOptions: string[];
+}
+
+interface PollEditorPanelState {
+  currentSettings: PollSettings;
+  currentOptions: string[];
+  validationWarning: string | undefined;
+}
+
+interface SavePayload {
+  alertPlayers: boolean;
+  expiryType: ExpiryType;
+  expiryValue: string;
+  multipleChoice: boolean;
+  options: string[];
+  servers: string;
+  title: string;
+}
+
+const createNewOptions = () => ['', ''];
+
+const processExpiry = (expiry: ExpiryOptions): ExpiryOptions => ({
+  ...expiry,
+  expiryValue: expiry.expiryValue?.trim(),
+});
+const processSettings = (settings: PollSettings) => {
+  const { expiry, title } = settings;
+  const processedExpiry = processExpiry(expiry);
+  return {
+    ...settings,
+    title: title.trim(),
+    expiry: processedExpiry,
+  };
+};
+const processOptions = (options: string[]) => options.map((option) => option.trim()).filter((option) => !!option);
+
+// rough yyyy-mm-dd checker, does not check for out-of-range days in shorter months
+const timestampRegex = /^\d{4}-[01]\d-[0-3]\d$/;
+const validateTimestamp = (timestamp: string) => {
+  if (!timestamp.match(timestampRegex)) {
+    return 'yyyy-mm-dd format required.';
+  }
+  return undefined;
+};
+const validate = (settings: PollSettings, options: string[]) => {
+  if (!settings.title) {
+    return 'Poll title is required.';
+  }
+  if (!settings.expiry?.expiryType) {
+    return 'Expiry setting is required.';
+  }
+  if (settings.expiry.expiryType !== 'never') {
+    if (settings.expiry.expiryValue === '') {
+      return 'Additional expiry information is required.';
+    }
+    if (settings.expiry.expiryType === 'timestamp') {
+      const timestampValidation = validateTimestamp(settings.expiry.expiryValue);
+      if (timestampValidation) {
+        return timestampValidation;
+      }
+    } else if (settings.expiry.expiryValue === '0') {
+      return 'Non-zero expiry time required.';
+    }
+  }
+  if (!settings.servers) {
+    return 'Server setting is required.';
+  }
+  if (options.length < 2) {
+    return 'At least 2 options required.';
+  }
+  return undefined;
+};
+
+/*
+ * N.B. On creation, this form was built to handle creating polls, with intent to expand
+ * to editing them at a later date. In its initial state, it only works for creating polls.
+ */
```

Comment:
Consider whether this should be done as part of it. Is this even needed (i.e. edit already existing poll via a form)?